Source/WebCore/page/csp/CSPViolationReportBody.cpp

```diff
@@ -0,0 +1,186 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CSPViolationReportBody.h"
+
+#include "ContentSecurityPolicyClient.h"
+#include "FormData.h"
+#include "SecurityPolicyViolationEvent.h"
+#include <wtf/IsoMallocInlines.h>
+
+namespace WebCore {
+
+using Init = SecurityPolicyViolationEventInit;
+
+WTF_MAKE_ISO_ALLOCATED_IMPL(CSPViolationReportBody);
+
+const AtomString& CSPViolationReportBody::cspReportType()
+{
+    static NeverDestroyed<AtomString> reportType { "csp-violation"_s };
+    return reportType;
+}
+
+CSPViolationReportBody::CSPViolationReportBody(Init&& init)
+    : ReportBody(ReportBodyType::CSPViolation)
+    , m_documentURL(WTFMove(init.documentURI))
+    , m_referrer(init.referrer.isNull() ? emptyString() : WTFMove(init.referrer))
+    , m_blockedURL(WTFMove(init.blockedURI))
+    , m_effectiveDirective(WTFMove(init.effectiveDirective))
+    , m_originalPolicy(WTFMove(init.originalPolicy))
+    , m_sourceFile(WTFMove(init.sourceFile))
+    , m_sample(WTFMove(init.sample))
+    , m_disposition(init.disposition)
+    , m_statusCode(init.statusCode)
+    , m_lineNumber(init.lineNumber)
+    , m_columnNumber(init.columnNumber)
+{
+}
+
+Ref<CSPViolationReportBody> CSPViolationReportBody::create(Init&& init)
+{
+    return adoptRef(*new CSPViolationReportBody(WTFMove(init)));
+}
+
+const AtomString& CSPViolationReportBody::type() const
+{
+    return cspReportType();
+}
+
+const String& CSPViolationReportBody::documentURL() const
+{
+    return m_documentURL;
+}
+
+const String& CSPViolationReportBody::referrer() const
+{
+    return m_referrer;
+}
+
+const String& CSPViolationReportBody::blockedURL() const
+{
+    return m_blockedURL;
+}
+
+const String& CSPViolationReportBody::effectiveDirective() const
+{
+    return m_effectiveDirective;
+}
+
+const String& CSPViolationReportBody::originalPolicy() const
+{
+    return m_originalPolicy;
+}
+
+const String& CSPViolationReportBody::sourceFile() const
+{
+    return m_sourceFile;
+}
+
+const String& CSPViolationReportBody::sample() const
+{
+    return m_sample;
+}
+
+SecurityPolicyViolationEventDisposition CSPViolationReportBody::disposition() const
+{
+    return m_disposition;
+}
+
+unsigned short CSPViolationReportBody::statusCode() const
+{
+    return m_statusCode;
+}
+
+unsigned long CSPViolationReportBody::lineNumber() const
+{
+    return m_lineNumber;
+}
+
+unsigned long CSPViolationReportBody::columnNumber() const
+{
+    return m_columnNumber;
+}
+
+Ref<FormData> CSPViolationReportBody::createReportFormDataForViolation(const CSPInfo& info, bool reportToFormat, bool isReportOnly,
```

Comment:
reportToFormat sounds like "reporting to format".
How about usesReportTo as in the caller?

Reply:
Will do!